changelog/selectiveUnittests.dd

```diff
@@ -0,0 +1,3 @@
+Only compile unittests in compiled modules
+
+The compiler was modified to only compile unittests in modules that are being compiled.  The most common impact is that unittests from druntime/phobos will not be included when compiling other programs that use these libraries.
```

Comment:
Actually...I should update this.  Turns out the unittests already weren't being compiled in (thanks @schveiguy for looking into this).  What's actually happening is that `version(unittest)` blocks are importing and instantiating templates that aren't instantiated without `-unittest`.  Then these instantiations are being moved to root modules.  With this new information, as far as I know this change shouldn't have any noticeable impact on users.

Reply:
So this just fixes sync between whether `unittest` blocks and `version(unittest)` blocks are being compiled, right?

Reply:
Actually no.  This is preventing template instances inside libraries from being moved to root modules when compiling with `-unittest`.  This is what's causing the compilation slowdown because the template instances that are being moved cause ALOT of phobos modules to be loaded and analyzed that wouldn't be otherwise.

`version(unittest)` blocks still get analyzed, however, @schveiguy is working on a phobos PR that should prevent those blocks from causing too much slowdown.  This PR along with the phobos PR should be everything we need to make `-unittest` compile just as fast as non `-unittest` compilations (when the app has no unittesting itself).